examples/prom/maxwell_global_rom.cpp

```diff
@@ -0,0 +1,548 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2023, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+//               libROM MFEM Example: parametric ROM for Maxwell equation (adapted from ex3p.cpp)
+//
+// Compile with: make maxwell_global_rom
+//
+// Description:  This example code uses MFEM and librom to define a projection-based
+//               reduced-order model for a simple electromagnetic diffusion
+//               problem corresponding to the second order definite Maxwell
+//               equation: curl curl E + E = f. The boundary condition is given by
+//               E x n = <given tangential field>. Here, we use a given exact
+//               solution E and compute the corresponding r.h.s. f. Each component
+//               of the vector field E is assumed to be a sin wave, with frequency
+//               controlled by the parameter kappa.
+//
+//               The example highlights three distinct ROM processes - the
+//               offline, merge, and online phases. The offline phase runs the full
+//               order model and stores the snapshot data in an HDF file. You
+//               can run as many offline phases as you wish to sample the
+//               parameter space. The merge phase reads all the snapshot files,
+//               builds a global reduced basis, and stores the basis in an HDF
+//               file. The online phase reads the basis, builds the ROM
+//               operator, solves the reduced order system, and lifts the
+//               solution to the full order space.
+//
+// For ROM (global rom):
+// Offline phase: maxwell_global_rom -offline -f 1.0 -id 0
+//                maxwell_global_rom -offline -f 1.1 -id 1
+//                maxwell_global_rom -offline -f 1.2 -id 2
+//
+// Merge phase:   maxwell_global_rom -merge -ns 3
+//
+// FOM solution:  maxwell_global_rom -fom -f 1.15
+//
+// Online phase:  maxwell_global_rom -online -f 1.15
+//
```

Comment:
Does it also work in parallel, like `poisson_global_rom`?

Reply:
I tested in parallel and confirmed that it works, so you can copy the comment from the Poisson example:
// This example runs in parallel with MPI, by using the same number of MPI ranks
// in all phases (offline, merge, fom, online).